src/Plugin/Hal.php

```diff
@@ -548,6 +562,10 @@ public function renderEntity(Entity $halEntity, $renderEntity = true)
             $entity = array();
         }
 
+        if ($this->maxDepth and $depth > $this->maxDepth) {
```

Comment:
We're seeing this from different angles and I think both should be supported

Given 
```
resource A includes B,(includes) C, D respectively so A.maxDepth = 3
resource B includes C, D, E B.maxDepth = 3;
```

I think you're suggesting this:
```
if C, D, E all .maxDepth 1;
actual depth starting from A = 4;

if C.maxDepth = 0
actual depth starting from A = 3

if B.maxDepth were changed to 0
actual depth starting from A = 2
```
and as you adjust C, D, E etc the math becomes more confusing.

and I say 
``` 
if C, D, E all .maxDepth 1;
actual depth starting from A = 3

if C.maxDepth = 0 || B.maxDepth
actual depth starting from A = 3
```

It seems reasonable to choose one or the other dictated by the top resource.  
So your math is
```
A.maxDepth = 3
A.elasticDepth = true;
```

And mine is 
```
A.maxDepth = 3;
A.elasticDepth = false;
```

Then comes some other hard math problems like 
```
A.maxDepth = 3; A.elasticDepth = true;
B.maxDepth = 0; B.elasticDepth = false;
```
Does this compute to A = 2? I think yes

```
A.maxDepth = 3; A.elasticDepth = false;
B.maxDepth = 0; B.elasticDepth = true;
```
A = 3 or 2?

From this conundrum I go back to assigning the true depth from the first, requested, resource.  And given the fuzzy math elasticDepth grafts to the logic I see non-elastic math as the sensible choice.